src/libratbag-data.c

```diff
@@ -124,22 +124,17 @@ init_data_hidpp10(struct ratbag *ratbag,
 	num = g_key_file_get_integer(keyfile, group, "DeviceIndex", &error);
 	if (num != 0 || !error)
 		data->hidpp10.index = num;
-	if (error)
-		g_error_free(error);
+	g_error_freep(&error);
```

Comment:
I'm sorry, but very much a NACK from me, this is likely going to mess up things long-term. Those free functions are "reserved" for exactly one use-case, messing with those means that, over time, we'll end up with something suboptimal and harder to read. Having said that, there's no need to even do this: `g_clear_error()` does exactly what we want :)

Alternatively, you could confine the error further in custom blocks:
```c
    int num;
    ...
    {
       g_autoptr(GError) error = NULL;
       num = g_get_file_get_integer(...)
       ...
   }
```

Reply:
`g_clear_error` seems nice, thanks! For some reason I keep getting lost in GLib's documentation, and I didn't see this function before...

Putting every key retrieval into blocks would be more modern in my opinion, but this pattern is not used in the code base much, so I think the `g_clear_error` fits better.